ngraph/test/backend/binary_convolution.in.cpp

```diff
@@ -0,0 +1,201 @@
+    copy_data(filters_tensor, filters);
+    auto result = backend->create_tensor(element::from<T_IN>(), outputs_shape);
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result}, {input_tensor, filters_tensor});
+    EXPECT_TRUE(test::all_close_f((outputs), read_vector<T_IN>(result), MIN_FLOAT_TOLERANCE_BITS));
+}
+
+// clang-format off
+// --------------------- 1D convolution ------------------------------------------
+NGRAPH_TEST(${BACKEND_NAME}, bin_convolution_1D_1batch_1channel_no_padding)
+{
+    const Strides strides{1};
+    const CoordinateDiff padding{0};
+    const Strides dilations{1};
+
+    const Shape inputs_shape{1, 1, 5};
+    const std::vector<float> inputs_conv{1.0f, -1.0f, -1.0f, 1.0f, -1.0f};
+    const std::vector<float> inputs_bin_conv{1.0f, 0.0f, 0.0f, 1.0f, 0.0f};
+
+    const Shape filters_shape{1, 1, 3};
+    const std::vector<float> filters_conv{1.0f, -1.0f, 1.0f};
+    const std::vector<uint8_t> filters_bin_conv{160}; // filters 1D: {1.0f, 0.0f, 1.0f}
+
+    const Shape outputs_shape{1, 1, 3};
+    const std::vector<float> outputs{1.0f, 1.0f, -3.0f};
+
+    BinaryConvolutionTest(
+        inputs_bin_conv,
+        inputs_shape,
+        filters_bin_conv,
+        filters_shape,
+        outputs,
+        outputs_shape,
+        strides,
+        padding,
+        dilations);
+
+    ConvolutionTest(
+        inputs_conv,
+        inputs_shape,
+        filters_conv,
+        filters_shape,
+        outputs,
+        outputs_shape,
+        strides,
+        padding,
+        dilations);
+}
+
+// --------------------- 2D convolution ------------------------------------------
+NGRAPH_TEST(${BACKEND_NAME}, bin_convolution_2D_1batch_1channel_no_padding)
+{
+    const Strides strides{1, 1};
+    const CoordinateDiff padding{0, 0};
+    const Strides dilations{1, 1};
+
+    const Shape inputs_shape{1, 1, 4, 4};
+    const std::vector<float> inputs_conv{1.0f, -1.0f, -1.0f, 1.0f,
+                                         1.0f, 1.0f, -1.0f, -1.0f,
+                                         -1.0f, -1.0f, -1.0f, 1.0f,
+                                         1.0f, -1.0f, 1.0f, 1.0f};
+
+    const std::vector<float> inputs_bin_conv{1.0f, 0.0f, 0.0f, 1.0f,
+                                             1.0f, 1.0f, 0.0f, 0.0f,
+                                             0.0f, 0.0f, 0.0f, 1.0f,
+                                             1.0f, 0.0f, 1.0f, 1.0f};
+
+    const Shape filters_shape{1, 1, 3, 3};
+    const std::vector<float> filters_conv{1.0f, -1.0f, 1.0f,
+                                          -1.0f, 1.0f, -1.0f,
+                                          1.0f, -1.0f, 1.0f};
+
+    //BinaryConvolution filters 2D:      {1.0f, 0.0f, 1.0f,
+    //                                    0.0f, 1.0f, 0.0f,
+    //                                    1.0f, 0.0f, 1.0f};
+    const std::vector<uint8_t> filters_bin_conv{170, 128};
```

Comment:
I would prefer bellow format for binary filter.  It's easier to read without calculator.
`
const std::vector<uint8_t> filters_bin_conv{0xAA, 0x80}; // 10101010 10000000
`

Reply:
Done